dynosaur_derive/src/lib.rs

```diff
@@ -217,8 +217,14 @@ fn mk_dyn_struct_impl_item(struct_ident: &Ident, erased_trait: &ItemTrait) -> To
                 _ => &args,
             };
 
-            quote! {
-                unsafe { &*self.ptr }.#ident(#(#args),*).await
+            if async_ {
```

Comment:
I would prefer to use the RPITIT form on the impl in both cases, for simplicity and because it should marginally improve compile times. If this turns out to be complicated for some reason though it can wait.